src/Endatix.Core/Entities/Identity/Role.cs

```diff
@@ -0,0 +1,36 @@
+using Ardalis.GuardClauses;
+using Endatix.Core.Infrastructure.Domain;
+
+namespace Endatix.Core.Entities.Identity;
+
+/// <summary>
+/// Represents the Role entity in the application domain.
+/// This class is not meant for direct persistence, but to handle domain logic related to roles.
+/// It encapsulates the core attributes and behaviors of a role within the application.
+/// Persistence implementation is done via the <see cref="AppRole"/>
+/// </summary>
+public sealed class Role : BaseEntity, IAggregateRoot
+{
+    public Role(
+        long id,
+        string name,
+        string? description
+        )
+    {
+        Guard.Against.Null(id);

```

Comment:
This check doesn't seem to work because id is not nullable.
In addition, I think we expect having only positive values for IDs and in other places we have guard clauses using `NegativeOrZero` for IDs. I suggest we use it here too.
The same applies for ID in User class.

Another question that comes to my mind is if we guard the ID, shouldn't we guard also name with NullOrWhiteSpace as having such a role/user is not correct.

Reply:
@dspetrov  - you are right. We should check for what's critical. I have added additional Guard clauses and fixed the Id one. ✅ resolved

Side note - Need to check at some point for more integrated validation for validating commands passed between MediatR. I have checked this - since we use Result, we can output Result.Invalid() with the validation errors before we process the items in the pipeline. Here is the task- https://github.com/endatix/endatix/issues/33